### ISSUE TITLE
Question about Filters

### ISSUE THREAD

**Issue by mdius** [concrete5-community-store/community_store#652]
Do you have a way of hiding filter options till a different option is selected?

Example:
I have Enclosures on my site, they come in 14 different sizes, I don't want 14 options shown unless they select the option enclosures.

If this is not possible, is there plans to do this?

**Comment by Mesuva** [comment]
Can you put some links to your example - I'm not sure I'm 100% following the challenge.

**Comment by mdius** [comment]
I have not set them all up yet.
But I'll try to explain better.
https://www.mdius.com/liquid-level-controls
This shows a LOT of filter options.

Would there be a way to maybe only show the Float Style filter only if you check Fiber Optic Switching under Switch Style?
This is how it looks right now.
![original](https://user-images.githubusercontent.com/28864099/156595554-3ef06348-8447-41e7-84dc-de71c33d4554.jpg)

This is how I'd like to see if I could do.
![original (would like)](https://user-images.githubusercontent.com/28864099/156595692-cfa9dbe0-76d3-4a24-b3c4-6404586b55dc.jpg)

Then when you click an option.
![original (then would)](https://user-images.githubusercontent.com/28864099/156595745-0a7a6b92-d1a3-480c-8d7a-f89d05c623af.jpg)

Did that help?

I did notice another issue, but I'll post that 1 later.

**Comment by Mesuva** [comment]
Yes, I think I'm following now.

The issue I'm seeing is how you define how each attribute filter interrelates.
In your case, I can see why you'd want the other options hidden first, as with these type of products it doesn't really make sense to select one of the lower filters until you've picked the main type of switching style.

But in other uses of the filter on other shops, there might not be this kind of thing at all, where a customer would just as likely filter by some option lower down the list.

I believe what you are after would be able to be achieved though with a custom block template for the filter block.
You'd put in some logic in the block template to output the first group, then only output the other groups of options _if_ the first group has at least one thing ticked.  So the short answer is to tackle it with PHP in a custom block template, instead of it being with javascript, or some feature that needs to be added to Community Store itself.

**Comment by mdius** [comment]
I think I'm following what you're suggesting.

Like when I said I saw another issue, it was that is was showing options with a Mini Float (0).
On line 53 of the Filter block, you have `<?php if ($show) { ?>`
I added `<?php if ($show && $count != 0) { ?>` to filter out the options with 0 counts.

It would be similar to this, only show the next block below if there are options available. That actually sounds better than what I was thinking. I'll let you know how it goes.

**Comment by mdius** [comment]
https://www.mytinytown.com/liquid-level-controls
So it seems I'm on the right track, I am just missing something. I just clicked the products with the highest amount, it seemed to be working till the 3rd group, then it gets weird, but still seems to kind of work. I'm not a fan of kind of though.

But then I question myself, would this be the best way, or would it be better to put them all in an accordion.
Click only the category you want to see. That would reduce the overwhelming selections, but still give all the options.

So, I'm going to give the accordion idea a try before I go any further. Also, with my fix above `<?php if ($show && $count != 0) { ?>` makes it so you cannot click 2 options in the same category.

**Comment by mdius** [comment]
Yeah, I think the accordion is the best idea as of now. I might pick back up on the other idea later.
https://www.mytinytown.com/liquid-level-controls

**Comment by mdius** [comment]
I have another question about filter/page list.
https://www.mytinytown.com/liquid-level-controls
On my test site I have added a Product Per Page feature that works, except:
I have is display 18 product by default, if I click 5 it works. Until I select a filter option then it defaults back to 18.  But if I select a Sort By option, it remains selected after selected a filter option. I narrowed it down to the word sort in the input select in blocks/community_product_list/view.php.
If I rename my Products Per Page input select name to "sort" it works, but then the Sort By obviously does not.
But I cannot figure out what triggers the sort option to stay.
I am guessing it is someplace in the Concrete core?
